Extension/BuiltInFunctionsExtension.php

```diff
@@ -183,7 +207,32 @@ public function currentUser()
     }
 
     /**
-     * Gets a random value between $min and $max
+     * Gets the datetime of the last login by user.
+     *
+     * @param null $userName
+     *
+     * @return \DateTime
+     *
+     * @throws \Exception
+     */
+    public function lastLogin($userName = null)
+    {
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        if (!is_null($userName)) {
+            $userRepository = $this->entityManager->getRepository($this->userClass);
+            $user = $userRepository->findOneBy(['username' => $userName]);
+
+            if (is_null($user)) {
+                throw new \Exception('User with username "'.$userName.'" could not be found.');
+            }
+        }
+
+        return $user->getLastLogin();
```

Comment:
How do you know this method exists. We cannot assume that this exists consumer of this bundle.